### DISCUSSION TITLE
4.0 veers to left before taking right turns at a Waypoint

### DISCUSSION THREAD

**Discussion by rts18** [iNavFlight/inav#7711] [General]
@breadoven @stronnag
Have either of you or anyone else experienced their models deviating to the left, as it approach's a waypoint?

I've noticed this effect with both MR and FW.  Both models ran the same waypoint test mission I used with previous releases, that **did not** experience this effect.

Its most pronounced when the next waypoint happens to be a right turn. Because the model will veer to the left at the waypoint and then make the right turn, to head to the next waypoint.
 
I have `nav_mc_wp_slowdown = off`. This has **always** been is my preferred method.

Unfortunately I don't have logs for these test missions. I only noticed the effect on three DVR play backs.

I did ponder that the models where experiencing a left crosswind.  Causing the navigation control to hook the model at the last second, to make the waypoint, before the turn.  But this didn't ring true on one mission, when the plane had a 150degree offset heading,  two waypoints later, no more than 500m away.

**Comment by stronnag** [comment]
I'm not sure it's only 4.0.0 that behaves strangely. Prior to 1.x (x < 8), tracking was perfect. Since then I've noticed acouple of things illustrated below:
* The curved track between WPs. Such as 10-11, 11-12, 1-2 in the image (not my flight).
![mission-wander](https://user-images.githubusercontent.com/158229/147847602-1f395bef-8d1d-48ac-ad60-f07e5f73134f.png)
* MR "Twitching" at a WP. The model will stop at the WP, rotate the wrong way before rotating the correct  way. The  flight exhibits this too, there's an non-public full replay at    https://youtu.be/1Yjmh2GKuzI which shows this affect (made before the user supplied the actual mission file). This is most apparent in the snippet below (WP6). IIRC the WP radius was set probably too low (maybe 1m).
https://user-images.githubusercontent.com/158229/147847747-8e570a72-0556-43f5-a571-759c14469968.mp4

I gave up worrying about such things in those ancient days. Missions always work, if not always elegantly.

**Comment by rts18** [comment]
@stronnag  Thanks for confirming the issue. As you noted, WP6 on your video was a classic case of what I've observed.
